wagtail/admin/templates/wagtailadmin/shared/side_panels/preview.html

```diff
@@ -12,18 +12,25 @@
 >
     <div class="w-preview__sizes">
         {% block size_buttons %}
-            <label class="w-preview__size-button">
-                {% icon name="mobile-alt" %}
-                <input type="radio" class="w-preview__proxy" name="preview-size" value="mobile" data-action="w-preview#togglePreviewSize" data-w-preview-target="size" data-device-width="375" data-default-size checked aria-label="{% trans 'Preview in mobile size' %}" />
-            </label>
-            <label class="w-preview__size-button">
-                {% icon name="tablet-alt" %}
-                <input type="radio" class="w-preview__proxy" name="preview-size" value="tablet" data-action="w-preview#togglePreviewSize" data-w-preview-target="size" data-device-width="768" aria-label="{% trans 'Preview in tablet size' %}" />
-            </label>
-            <label class="w-preview__size-button">
-                {% icon name="desktop" %}
-                <input type="radio" class="w-preview__proxy" name="preview-size" value="desktop" data-action="w-preview#togglePreviewSize" data-w-preview-target="size" data-device-width="1280" aria-label="{% trans 'Preview in desktop size' %}" />
-            </label>
+            {% for size in object.preview_sizes %}
+                <label class="w-preview__size-button">
+                    {% icon name=size.icon %}
+                    <input
+                        type="radio"
+                        class="w-preview__proxy"
+                        name="preview-size"
+                        value="{{ size.name }}"
+                        data-action="w-preview#togglePreviewSize"
+                        data-w-preview-target="size"
+                        aria-label="{{ size.label }}"
+                        data-device-width="{{ size.device_width|stringformat:'s' }}"
+                        {% if object.default_preview_size == size.name %}
+                            data-default-size
```

Comment:
I previously added this so we could keep remember the mobile size which is used to display the "Preview not available" screen regardless of the currently selected device size. Otherwise, the message (which is inside the iframe) would be scaled according to the selected device size as if we're doing a preview, which might be unreadable.

<img width="418" alt="image" src="https://github.com/user-attachments/assets/1e4d9f84-d0be-40f6-bebb-b26f5d3d7648">



Now that the default size might be something else, I guess we can drop this data attribute and just rely on `checked` to set the initial state. Then, we'll just hardcode the preferred size for the "unavailable" screen in the controller code.

Reply:
I've added a commit to drop `data-default-size` and also fix this issue

<img width="392" alt="image" src="https://github.com/user-attachments/assets/070b886b-dccc-449d-94b2-ad193c6ff33c">